src/snowflake/snowpark/modin/plugin/_internal/frame.py

```diff
@@ -691,6 +692,22 @@ def ensure_row_count_column(self) -> "InternalFrame":
             index_column_snowflake_quoted_identifiers=self.index_column_snowflake_quoted_identifiers,
         )
 
+    def materialize_and_cache(self) -> "InternalFrame":
```

Comment:
hm I feel a proper name here might be `persist_to_temporary_table`. Materialize is often used in a different context and this may be misleading. Would you remind renaming?

Reply:
Makes sense!